pkg/exchange/coinbase/stream_handlers.go

```diff
@@ -80,31 +81,68 @@ func (s *Stream) handleConnect() {
 	if len(s.Subscriptions) == 0 {
 		return
 	}
-
+	// bridge bbgo channels to coinbase channels
+	// auth required: level2, full, user
 	subProductsMap := make(map[types.Channel][]string)
 	for _, sub := range s.Subscriptions {
-		// rfqMatchChannel allow empty symbol
-		if sub.Channel != rfqMatchChannel && sub.Channel != statusChannel && len(sub.Symbol) == 0 {
-			continue
+		localSymbol := toLocalSymbol(sub.Symbol)
+		switch sub.Channel {
+		case types.BookChannel:
+			// bridge to level2 channel, which provides order book snapshot and book updates
+			logStream.Infof("bridge %s to level2 channel (%s)", sub.Channel, sub.Symbol)
+			subProductsMap[level2Channel] = append(subProductsMap[level2Channel], localSymbol)
+		case types.MarketTradeChannel:
+			// full/user channels both provide feeds on orders and trades
+			// full: all orders/trades on Coinbase Exchange
+			// user: orders/trades belong to the user
+			var localChannel types.Channel
+			if s.PublicOnly {
```

Comment:
MarketTradeChannel is only available in the publicOnly mode

Reply:
Done.